src/utils/FixedPointMathLib.sol

```diff
@@ -9,6 +9,7 @@ library FixedPointMathLib {
                     SIMPLIFIED FIXED POINT OPERATIONS
     //////////////////////////////////////////////////////////////*/
 
+    uint256 internal constant MAX_UINT256 = 2**256 - 1;
```

Comment:
can we use type(uint256).max here instead

Reply:
I think that it is impossible in assembly

Reply:
Yes not possible, unfortunately. We can use `not(0)` to replace it though

Reply:
EDIT: the gas seems to increase a bit (see below). I left the config at 1M optimizer runs.

<img width="1383" alt="Screenshot 2022-10-16 at 14 15 55" src="https://user-images.githubusercontent.com/44097430/196034910-0bf42ac3-a4cc-4b5f-89c3-6ca2d9d29635.png">

Reply:
i just meant writing 

```
uint256 internal constant MAX_UINT256 = type(uint256).max;
```

haha

Reply:
welp alright

Reply:
Yeah clearly